protocols/grpc/agent.proto

```diff
@@ -170,24 +174,17 @@ message Route {
 	string device = 3;
 }
 
-enum UpdateType {
-	None = 0;
-	AddIP = 1;
-	RemoveIP = 2;
-	Name = 4;
-	MTU = 8;
-}
-
 message AddInterfaceRequest {
-	Interface interface = 1;
+	string interface_id = 1;
```

Comment:
We do not need to pass an interface ID for a network interface. As discussed, the MAC address should be sufficient to unambiguously identify a network interface.

Reply:
MAC is also a field allowed to be changed.
for example, future container migration via vm migration. Although we can remove&add to achieve it.

Reply:
I hope every obj have its name which is from and can be randomly generated by upper layers

Reply:
@laijs the reason to use macaddress as an identifier is that it maps to the same value both outside and inside the VM. Hence we do not need to know what the predicted interface name within the VM  needs to be. That allows the name within the VM to be changed to refect the desired name of the interface within the virtualmachine. For example the VM kernel/udev rules may generate enp0s1 and we may want to change it to eth0. So in that case using the name as an identifier will not work for the first time configuration. Hence using hardware based identifiers like mac or pci b/d/f which are fully predictable makes more sense.

Reply:
@laijs MAC can be modified, but different VM can't own same MAC, right? Or the network and route wont work.

@mcastelino Yes, I agree that MAC is a better identifier than name, you're right.